### ISSUE TITLE
Support for Visual Studio "Object File Name" ("/Fo")

### ISSUE THREAD

**Issue by imallett** [premake/premake-core#1270]
In larger projects, files can have the same name, so dumping compiled files all in the same directory, as-is the inexplicable Visual Studio default, doesn't work. For this reason, I always set the "C/C++" -> "Object File Name" to something like "$(IntDir)/%(RelativeDir)/".

As far as I can tell, premake does not currently support setting this. Moreover, it cannot be worked around by passing the flag "manually" via `buildoptions(...)` since this produces an unsurpressable D9025 warning (the global property pages define a different value) and also fails to build correctly (which are separate issues with Visual Studio, but).

Please add this feature. Thank you.

**Comment by samsinsane** [comment]
I assume that using [`objdir`](https://github.com/premake/premake-core/wiki/objdir) wouldn't suffice in this case?

**Comment by imallett** [comment]
I don't believe so, no. Specifically, `/Fo` specifies the path for each individual file (which can be used, as in my case, to specify the relative directory), whereas `objdir(...)` only specifies their common output directory.

**Comment by ratzlaff** [comment]
Premake should detect this use case and specify unique object names for those files. I personally have generated projects containing different cpp files of the same name before. What version of premake are you using? Which visual studio generator are you using?

**Comment by imallett** [comment]
Although I posed this as problems with the build, admittedly I was not testing on a project where there actually are duplicate filenames. I can confirm that premake detects this case on a different project. (Premake version: trunk as of a few days ago. VS version: 2019.)

I suppose some remaining technical motivations for this are then:
- Premake's solution, tacking a number on the end, causes problems with programs outside of premake that expect object files to have well-defined places and filenames (e.g. CUDA+OptiX).
- Similarly, it affects user tools (e.g. objdump, etc.). If there are 5 different numbers (as is the case for this current project), it's impossible to tell which is which for e.g. disassembly.
- Makes it harder to organize the build directory.
- Bloats the source control for the project files, because a special filename is specified for each file.

There's also the non-technical motivation that users "should" be able to specify how their object files are written, and it's "elegant" to have the build directory's structure mirror the source directory's (wherefore I don't understand why this isn't Visual Studio's default).

(It should be noted that that default can be changed globally in Visual Studio by changing "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/MSBuild/Microsoft/VC/v160/Microsoft.Cl.Common.props:78", which appears to be where the value currently comes from, to "$(IntDir)\\%(RelativeDir)", with no trailing slash. This is, of course, brittle, and doesn't scale to other users. Besides which there are big warnings at the top about modifying the file, and indeed doing it incorrectly breaks e.g. CMake at a system-wide level.)

**Comment by p0358** [comment]
One more caveat of not changing the "Object File Name":

if you create a new file in your Visual Studio project manually temporarily (as it'd normally work), it'd then not gonna work due to obj name clash -- you'll need to re-run your premake scripts right away in such case

(not saying it's a big deal, but worth noting, perhaps will be useful to someone in the future...)
